Assets/Scripts/Character/HealthSystem.cs

```diff
@@ -0,0 +1,52 @@
+using UnityEngine;
+
+public class HealthSystem : MonoBehaviour
+{
+    [Header("Health Settings")]
+    public float maxHealth = 100.0f;
+    public float currentHealth;
+
+    [Header("Damage Settings")]
+    public LayerMask enemyLayers;
+    public float damageOnCollision = 10.0f;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        currentHealth = maxHealth;
+    }
+
+    public void TakeDamage(float damage)
+    {
+        currentHealth -= damage;
+
+        if (currentHealth <= 0)
+        {
+            Die();
+        }
+    }
+
+    private void OnTriggerEnter(Collider other)
```

Comment:
why is the health system doing damage? shouldn't this happen in the attack state?